include/vara/Feature/FeatureModel.h

```diff
@@ -0,0 +1,222 @@
+        }
+      }
+      llvm::outs() << "\n";
+    }
+  }
+
+  static bool
+  skip(std::pair<FMFeature *, FMFeature *> Edge,
+       const llvm::SmallSet<std::pair<FMFeature *, FMFeature *>, 10> &Visited) {
+    for (const auto &P : Visited) {
+      if (P.first == Edge.first && P.second == Edge.second) {
+        return true;
+      }
+    }
+    return false;
+  }
+};
+} // namespace vara
+
+inline std::ostream &operator<<(std::ostream &Out,
+                                const vara::FMFeature *FMFeature) {
+  FMFeature->print(Out);
+  return Out;
+}
+
+inline llvm::raw_ostream &operator<<(llvm::raw_ostream &Out,
+                                     const vara::FMFeature *FMFeature) {
+  FMFeature->print(Out);
+  return Out;
+}
+
+namespace llvm {
+
+//===----------------------------------------------------------------------===//
+//                     (Dot)GraphTraits for FeatureModel
+//===----------------------------------------------------------------------===//
+
+template <> struct GraphTraits<vara::FeatureModel *> {
+  using NodeRef = typename vara::FMFeature *;
+
+  static NodeRef getEntryNode(const vara::FeatureModel *FM) {
+    return FM->getRoot();
+  }
+
+  using nodes_iterator = typename vara::FeatureModel::FeatureModelIter;
+
+  static nodes_iterator nodes_begin(vara::FeatureModel *FM) {
+    return FM->begin();
+  }
+
+  static nodes_iterator nodes_end(vara::FeatureModel *FM) { return FM->end(); }
+
+  using ChildIteratorType = typename vara::FMFeature::feature_iterator;
+
+  static ChildIteratorType child_begin(NodeRef N) { return N->begin(); }
+
+  static ChildIteratorType child_end(NodeRef N) { return N->end(); }
+
+  static size_t size(vara::FeatureModel *FM) { return FM->size(); }
+};
+
+template <>
+struct DOTGraphTraits<vara::FeatureModel *> : public DefaultDOTGraphTraits {
+  explicit DOTGraphTraits(bool IsSimple = false)
+      : DefaultDOTGraphTraits(IsSimple) {}
+
+  static std::string getGraphName(const vara::FeatureModel *FM) {
+    return "Feature model for " + FM->getName().str();
+  }
+
+  std::string getNodeLabel(const vara::FMFeature *Node,
+                           const vara::FeatureModel *FM) {
+    std::stringstream S;
+    S << Node->getName();
+    return S.str();
+  }
+
+  static std::string getNodeAttributes(const vara::FMFeature *Node,
+                                       const vara::FeatureModel *FM) {
+    return "";
+  }
+
+  std::string getEdgeAttributes(const vara::FMFeature *Node,
+                                vara::FMFeature::feature_iterator I,
+                                const vara::FeatureModel *FM) {
+    std::stringstream S;
+    S << "arrowhead=\"";
+    if ((*I)->isOptional()) {
+      S << 'o';
+    }
+    S << "dot\" ";
+    return S.str();
+  }
+
+  template <typename GraphWriter>
+  static void addCustomGraphFeatures(vara::FeatureModel *FM, GraphWriter &W) {
+    llvm::SmallSet<std::pair<vara::FMFeature *, vara::FMFeature *>, 10> SkipE,
+        SkipI, SkipA;
```

Comment:
Could you please spell out the names, E/I/A makes it quite hard to read for me.
`SkipExcludes` would make it easier for me :grin: